### PR TITLE
Upgrade Edge-Core 0.15.0 And Enable Maestro GRM feature For Pelion Edge 2.2

### PR DESCRIPTION
**Pull request by costanic**

### CODE REVIEW

**Review by ygoyal18 — APPROVED** [review]
LGTM

**Comment by costanic** [comment]
> LGTM

ok thanks @ygoyal18.  Not sure what the CI error is, because it builds and works in my dev environment.  Once I figure that out, I'll ping you again if needed.

File: snap/snapcraft.yaml [hunk 1]
```diff
@@ -223,7 +223,7 @@ parts:
     edge-core:
       plugin: cmake
       source: https://github.com/ARMmbed/mbed-edge.git
-      source-tag: 0.13.0
+      source-tag: 0.15.0
```
Comment (ygoyal18) [inline snap/snapcraft.yaml:226]:
```suggestion
      source-tag: 0.16.1
```
@costanic New version of edge-core is available so if we are upgrading then let's go to latest one? Not sure if we will have to regenerate all the patches to work with new version or not.

Comment (costanic) [reply snap/snapcraft.yaml:226]: @ygoyal18 This PR, plus an [upcoming PR from the pe-2.2](https://github.com/PelionIoT/snap-pelion-edge/tree/pe-2.2) branch encompasses an effort to upgrade snap to PE2.2.  Is PE2.2 running with 0.16.1?

Comment (ygoyal18) [reply snap/snapcraft.yaml:226]: You are right, 2.2 is on EC 0.15. But 2.3 will be available by end of this month and that will take us to 0.16.1. We will again we out of sync and as we haven't yet planned to support Parsec in snap, we won't get chance to upgrade edge-core to latest version. My suggestion is to take latest version if we are in this space and doing the leg work to upgrade edge-core version.

Comment (costanic) [reply snap/snapcraft.yaml:226]: @ygoyal18 Do you have a list of other component versions for PE2.3?  I.e., which version of edge-proxy, maestro, pe-utils, dockerd, etc.?  Please see [this branch](https://github.com/PelionIoT/snap-pelion-edge/commits/pe-2.2) for current status of upgrading snap components for 2.2, which was planned to be merged after this PR.  Which of those components will need to be upgraded again for 2.3 and which are already at the correct version?

**Comment by costanic** [comment]
Also @ygoyal18 where can I find a Changelist for 0[.16.1?](https://github.com/PelionIoT/mbed-edge/releases/tag/0.16.0)  The change from 0.13.0 to 0.15.0 was drastic as you can see in this PR, so I need to understand the additional effort to upgrade from 0.15.0 to 0.16.1.

**Comment by ryannowarm** [comment]
trying to re-run CI to see if we can move this along

**Review by mray190 — APPROVED** [review]
LGTM other than CI

**Review by CristianPrundeanuARM — APPROVED** [review]

**Review by johnmichaelwelsh — APPROVED** [review]
Looks ok.

**Comment by ygoyal18** [comment]
> Also @ygoyal18 where can I find a Changelist for [0.16.1](https://github.com/PelionIoT/mbed-edge/releases/tag/0.16.1)? The change from 0.13.0 to 0.15.0 was drastic as you can see in this PR, so I need to understand the additional effort to upgrade from 0.15.0 to 0.16.1.

@costanic Here is the changelog for 0.16 release - https://github.com/PelionIoT/mbed-edge/blob/master/CHANGELOG.md#release-0160-2021-3-15.

File: snap/snapcraft.yaml (hunk 1 above)
Comment (ygoyal18) [reply snap/snapcraft.yaml:226]: AFAIK, today there is no easy way to get that information out of meta-pelion-edge and meta-mbed-edge. I remember we discussed with @ryannowarm to have a version yaml file or similar in our packages which will give us that information in one shot. I forgot what we called it. 

From comparing the dev with master branch of Yocto Pelion Edge I can only see mbed-edge being upgraded to 0.16.1 and mbed-edge-examples to 0.16.0. Rest all the packages are at the same version as 2.2.

**Comment by costanic** [comment]
> > Also @ygoyal18 where can I find a Changelist for [0.16.1](https://github.com/PelionIoT/mbed-edge/releases/tag/0.16.1)? The change from 0.13.0 to 0.15.0 was drastic as you can see in this PR, so I need to understand the additional effort to upgrade from 0.15.0 to 0.16.1.
> 
> @costanic Here is the changelog for 0.16 release - https://github.com/PelionIoT/mbed-edge/blob/master/CHANGELOG.md#release-0160-2021-3-15.

Thanks @ygoyal18, that looks fairly straight-forward and I've done [some work towards 0.16.1](https://github.com/pelioniot/snap-pelion-edge/tree/edge-core-0.16.1) and got it to compile, but since we're out of time in our sprint I'd like to go ahead and merge this as-is so that we can meet our sprint goal.  Then, as soon as I have time to dev-test the 0.16.1 changes I'll create a PR for it.  Sound ok?

File: snap/snapcraft.yaml (hunk 1 above)
Comment (costanic) [reply snap/snapcraft.yaml:226]: ok thanks @ygoyal18, I can just look at the metas until we have a PE version index or whatever we called it.

**Pull request merged**